share/sql/share.go

```diff
@@ -537,6 +338,266 @@ func (m *shareMgr) UpdateReceivedShare(ctx context.Context, recvShare *collabora
 	return rs, nil
 }
 
+// Exported functions below are not part of the CS3-defined API, but are used by cernboxcop
+
+func (m *shareMgr) GetShareUnfiltered(ctx context.Context, ref *collaboration.ShareReference) (*model.Share, error) {
+	share, err := m.getShare(ctx, ref, false)
+	if err != nil {
+		return nil, err
+	}
+
+	return share, nil
+}
+
+// TransferShare transfers a share to a new owner. Only to be used for shares in projects.
+func (m *shareMgr) TransferShare(ctx context.Context, ref *collaboration.ShareReference, newOwner string) (*collaboration.Share, error) {
+	if newOwner == "" {
+		return nil, errors.New("Must pass a non-nil owner")
+	}
+
+	share, err := m.getEmptyShareByRef(ctx, ref)
+	if err != nil {
+		return nil, err
+	}
+
+	res := m.db.Model(&share).Update("uid_owner", newOwner)
+	if res.Error != nil {
+		return nil, res.Error
+	}
+
+	return m.GetShare(ctx, ref)
+}
+
+// Move share moves a share to a new location, also updating its owner. It is the reponsibility of the caller to ensure that `newOwner`
+// corresponds to the owner of `newLocation`
+func (m *shareMgr) MoveShare(ctx context.Context, ref *collaboration.ShareReference, newLocation *provider.ResourceId, newOwner string) (*collaboration.Share, error) {
+	if newOwner == "" {
+		return nil, errors.New("Must pass a non-nil owner")
+	}
+
+	if newLocation.OpaqueId == "" || newLocation.StorageId == "" {
+		return nil, errors.New("Must pass a non-nil location")
+	}
+
+	share, err := m.getEmptyShareByRef(ctx, ref)
+	if err != nil {
+		return nil, err
+	}
+
+	res := m.db.Model(&share).Update("uid_owner", newOwner).Update("inode", newLocation.OpaqueId).Update("instance", newLocation.StorageId)
+	if res.Error != nil {
+		return nil, res.Error
+	}
+
+	return m.GetShare(ctx, ref)
+}
+
+func (m *shareMgr) getPath(ctx context.Context, resID *provider.ResourceId) (string, error) {
```

Comment:
This function here seems a bit strange.. Can't we get it from somewhere else?

Reply:
I don't think so because it's just a wrapper around a call to `GetPath` in the gateway